### PR TITLE
Fix warnings and avoid processing names when not required

### PR DESCRIPTION
**Pull request by bcsanches**
Hello,

first of all thanks for this excellent library.

I did some small fixes for annoying warnings on Windows version and I hope they can be useful to you and others.

Thanks

### CODE REVIEW

File: src/lfwatch_win32.cpp
Comment (Twinklebear) [inline src/lfwatch_win32.cpp]: For the unused parameter warning, we can just remove the parameter name in the `watch_callback` definition:

```c++
 void CALLBACK watch_callback(DWORD err, DWORD, LPOVERLAPPED overlapped) {
```

File: src/lfwatch_win32.cpp
Comment (Twinklebear) [inline src/lfwatch_win32.cpp:88]: Can we also keep the 0 initialization here? I think the null terminator shouldn't be lost since it's added on line 86 since we make `wfname` 1 character bigger, but probably best to keep this 0 initialized.

**Review by Twinklebear — CHANGES_REQUESTED** [review]
Thanks @bcsanches ! I added a few smaller notes to tweak, I'm fine to make those after merging or can leave it to you if you'd like

File: src/lfwatch_win32.cpp
Comment (bcsanches) [reply src/lfwatch_win32.cpp]: Yes, you are right, that solution is better.

File: src/lfwatch_win32.cpp
Comment (bcsanches) [reply src/lfwatch_win32.cpp:88]: You can keep, makes no harm, but the wcstombs_s guarantees that it will put a \0 at the string end, thats why I removed the 0 initialization. 

I am not a big fan of all those changes that microsoft did to the standard library. Another solution is to modify the cmake file to include the macro _CRT_SECURE_NO_WARNINGS and keep the code unchanged. 

What do you think?

**Comment by bcsanches** [comment]
I changed the watch_callback definition and removed the unused parameter name for a better fix.

File: src/lfwatch_win32.cpp
Comment (Twinklebear) [reply src/lfwatch_win32.cpp:88]: Lets keep it as you have in the PR then, since it'll always add the null terminator

**Review by Twinklebear — APPROVED** [review]
Looks good!

**Comment by Twinklebear** [comment]
Thanks @bcsanches !

**Comment by bcsanches** [comment]
Thank you!

**Pull request merged**
